packages/core/contracts/l2/polygon/OracleChildTunnel.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./tunnel/FxBaseChildTunnel.sol";
+import "../../oracle/interfaces/OracleAncillaryInterface.sol";
+import "../../oracle/interfaces/RegistryInterface.sol";
+import "./OracleBaseTunnel.sol";
+
+/**
+ * @title Adapter deployed on L2 to give financial contracts the ability to trigger cross-chain price requests to the
+ * L1 DVM. Also has the ability to receive published prices from L1. This contract can be treated as the "DVM" for this
+ * network, because a calling contract can request and access a resolved price request from this contract.
+ * @dev The intended client of this contract is an OptimisticOracle on L2 that needs price
+ * resolution secured by the DVM on L1.
+ */
+contract OracleChildTunnel is OracleBaseTunnel, OracleAncillaryInterface, FxBaseChildTunnel {
+    constructor(address _fxChild, address _finderAddress)
+        OracleBaseTunnel(_finderAddress)
+        FxBaseChildTunnel(_fxChild)
+    {}
+
+    // This assumes that the local network has a Registry that resembles the Mainnet registry.
+    modifier onlyRegisteredContract() {
+        RegistryInterface registry = RegistryInterface(finder.getImplementationAddress(OracleInterfaces.Registry));
+        require(registry.isContractRegistered(msg.sender), "Caller must be registered");
+        _;
+    }
+
+    /**
+     * @notice This should be called to bridge a price request to Mainnet.
+     * @dev Can be called only by a Registered contract that is allowed to make DVM price requests. Will mark this
+     * price request as Requested, and therefore able to receive the price resolution data from L1. Emits a message
+     * that will be included in regular checkpoint of all Matic transactions to Ethereum.
+     * @param identifier Identifier of price request.
+     * @param time Timestamp of price request.
+     * @param ancillaryData extra data of price request.
+     */
+    function requestPrice(
+        bytes32 identifier,
+        uint256 time,
+        bytes memory ancillaryData
+    ) public override onlyRegisteredContract() {
+        bytes32 priceRequestId = _encodePriceRequest(identifier, time, ancillaryData);
+        Price storage lookup = prices[priceRequestId];
+        if (lookup.state != RequestState.NeverRequested) {
+            // Clients expect that `requestPrice` does not revert if a price is already requested, so return gracefully.
+            return;
+        } else {
+            _requestPrice(identifier, time, ancillaryData);
+
+            // Initiate cross-chain price request:
+            // TODO: Can we pack more information into this request? We could try to check if the requester is an
+            // OptimisticOracle and pull price request metadata from that?
+            _sendMessageToRoot(abi.encode(identifier, time, ancillaryData));
```

Comment:
Reinis brought up in the chainbridge UMIP review that we should ideally publish information such as fees and original requester address to Mainnet to aid DVM voters. However, there is no obvious way to do this because the `OptimisticOracle` expects to call `requestPrice(bytes32,uint25,bytes)` according to the `OracleInterface`, so we can't pack more information into this call without creating a new interface and updating the `OptimisticOracle`. 

We _could_ assume that the requester was an OO and query data from the caller of this `requestPrice` method. However, see the note in `OracleRootTunnel` how `_processMessageFromChild` will NOT emit an event on Ethereum because that internal function is expected to be called via a System call that does not send a transaction.

Reply:
would `ancillaryData` not help a bit with this? it would be possible to append more info to it without affecting the original intention of the data, no?

Reply:
> would `ancillaryData` not help a bit with this? it would be possible to append more info to it without affecting the original intention of the data, no?

Really good point. We'd have to modify the OptimisticOracle implementation that we deploy to Polygon to stamp additional information like `finalFee`, `bond`, `requester` etc. But that's ok because we do need to deploy the OO again to Polygon!

We could create a new `OptimisticOracleL2` contract for this purpose. Thoughts?

Reply:
Or maybe we just make a follow up PR that adds a lot information to the `OptimisticOracle` stamped ancillary data, and deploy that new version to Polygon?

Reply:
ye I think that's totally reasonable. when we were thinking through ancillary data the key pattern was this ability to "stamp" data from a number of sets the data is passed through. an L2 bridge is a great example of where this stamping would work really well!

Reply:
Also just noticed that the OO does stamp in `requester` address so it could make sense to add extra data

Reply:
@chrismaree @mrice32 here's a change to the `OptimisticOracle` that we should make to help voters with decoding the stamped ancillary data. Additionally, consider what other pieces of data we might want to stamp to the DVM: ([PR](https://github.com/UMAprotocol/protocol/pull/3061))